### PR TITLE
Add a workflow example for tekton CRs

### PR DESCRIPTION
**Pull request by ydayagi**
[FLPATH-216](https://issues.redhat.com//browse/FLPATH-216) https://issues.redhat.com/browse/FLPATH-216

### CODE REVIEW

File: workflow-examples/src/test/java/com/redhat/parodos/examples/tekton/TektonWorkFlow.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+			WorkFlowRequestDTO workFlowRequestPipeline = new WorkFlowRequestDTO().projectId(testProject.getId())
+					.workFlowName(workflowName).works(Arrays.asList(workPipeline));
+			WorkFlowRequestDTO workFlowRequestPipelinerun = new WorkFlowRequestDTO().projectId(testProject.getId())
+					.workFlowName(workflowName).works(Arrays.asList(workPipelineRun));
+
+			WorkFlowRequestDTO workFlowRequests[] = new WorkFlowRequestDTO[] { workFlowRequestNamespace,
+					workFlowRequestTask1, workFlowRequestTask2, workFlowRequestPipeline, workFlowRequestPipelinerun };
+
+			// 3 - Execute WorkF
+			//
```
Comment (pkliczewski) [inline workflow-examples/src/test/java/com/redhat/parodos/examples/tekton/TektonWorkFlow.java:153]: Is this expected to be empty?

**Review by pkliczewski — COMMENTED** [review]
Please add an entry in the documentation section on our website.

File: workflow-examples/src/test/java/com/redhat/parodos/examples/tekton/TektonWorkFlow.java (hunk 1 above)
Comment (ydayagi) [reply workflow-examples/src/test/java/com/redhat/parodos/examples/tekton/TektonWorkFlow.java]: fixed

**Comment by ydayagi** [comment]
> Please add an entry in the documentation section on our website.

@pkliczewski I have a story https://issues.redhat.com/browse/FLPATH-402 for documenting all prebuilt tasks. I though it could be good to finish with them and then see which format is more suitable. I have just the existing tasks in workflow-example that I need to move and then I will start documentation. Is that OK?

File: workflow-examples/src/test/java/com/redhat/parodos/examples/tekton/TektonWorkFlow.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
+package com.redhat.parodos.examples.tekton;
```
Comment (masayag) [inline workflow-examples/src/test/java/com/redhat/parodos/examples/tekton/TektonWorkFlow.java:1]: Shouldn't this be added to the `integration-tests` module?

File: workflow-examples/src/test/java/com/redhat/parodos/examples/tekton/TektonWorkFlow.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+		ApiClient defaultClient = Configuration.getDefaultApiClient();
+
+		defaultClient.addDefaultHeader(HttpHeaders.AUTHORIZATION, "Basic " + CredUtils.getBase64Creds("test", "test"));
+
+		try {
+			ProjectResponseDTO testProject = getProjectAsync(defaultClient, projectName, projectDescription);
+
+			// GET workflow DEFINITIONS
+			WorkflowDefinitionApi workflowDefinitionApi = new WorkflowDefinitionApi(defaultClient);
+			List<WorkFlowDefinitionResponseDTO> simpleSequentialWorkFlowDefinitions = workflowDefinitionApi
```
Comment (masayag) [inline workflow-examples/src/test/java/com/redhat/parodos/examples/tekton/TektonWorkFlow.java:49]: pls replace with 
```
TestComponents components = new WorkFlowTestBuilder().withDefaultProject().withWorkFlowDefinition(WORKFLOW_NAME)
				.build();
```
which does that already.
See any other workflow in the integration-tests

File: workflow-examples/src/test/java/com/redhat/parodos/examples/tekton/TektonWorkFlow.java [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+			// GET WORKFLOW DEFINITION BY Id
+			WorkFlowDefinitionResponseDTO simpleSequentialWorkFlowDefinition = workflowDefinitionApi
+					.getWorkFlowDefinitionById(simpleSequentialWorkFlowDefinitions.get(0).getId());
+
+			// EXECUTE WORKFLOW
+			WorkflowApi workflowApi = new WorkflowApi();
+
+			// 1 - Define WorkRequests
+			String kubeconfigJson = Files.readString(new File("kubeconfig.json").toPath());
+
```
Comment (masayag) [inline workflow-examples/src/test/java/com/redhat/parodos/examples/tekton/TektonWorkFlow.java:62]: this is hard to view. you can use a similar way as in prebuilt-tasks/src/test/java/com/redhat/parodos/tasks/deploy/DeployApplicationTaskTest.java to define the json as:
```
			// @formatter:off
			String deploymentContent = """
apiVersion: apps/v1
kind: Deployment
metadata:
  name: my-deployment
spec:
  replicas: 3
  selector:
    matchLabels:
      app: my-app
  template:
    metadata:
      labels:
        app: my-app
    spec:
      containers:
      - name: my-container
        image: my-image:latest""";
			// @formatter:on
```

File: workflow-examples/src/test/java/com/redhat/parodos/examples/tekton/TektonWorkFlow.java [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+import org.junit.Test;
+
+import org.springframework.http.HttpHeaders;
+
+import static com.redhat.parodos.sdkutils.WorkFlowServiceUtils.getProjectAsync;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+public class TektonWorkFlow {
```
[the remark this answers is not in the document]
Comment (ydayagi) [reply workflow-examples/src/test/java/com/redhat/parodos/examples/tekton/TektonWorkFlow.java:28]: @masayag these files are intentionally without the suffix 'Test'. Look at all the other files. There is a good reason for that. It is not a test. This is how the original authors implemented it.

File: workflow-examples/src/test/java/com/redhat/parodos/examples/tekton/TektonWorkFlow.java (hunk 4 above)
Comment (ydayagi) [reply workflow-examples/src/test/java/com/redhat/parodos/examples/tekton/TektonWorkFlow.java:62]: I will try

File: workflow-examples/src/test/java/com/redhat/parodos/examples/tekton/TektonWorkFlow.java (hunk 3 above)
Comment (ydayagi) [reply workflow-examples/src/test/java/com/redhat/parodos/examples/tekton/TektonWorkFlow.java:49]: I will try

Comment (ydayagi) [reply workflow-examples/src/test/java/com/redhat/parodos/examples/tekton/TektonWorkFlow.java:49]: I can not use it. It is from a different module that the examples should not rely on. The examples are equivalent to a customer's project/module. Customer should not depend on our integration tests

File: workflow-examples/src/test/java/com/redhat/parodos/examples/tekton/TektonWorkFlow.java (hunk 4 above)
Comment (ydayagi) [reply workflow-examples/src/test/java/com/redhat/parodos/examples/tekton/TektonWorkFlow.java:62]: fixed

**Review by pkliczewski — COMMENTED** [review]
lgtm. @masayag what do you think?

**Comment by masayag** [comment]
> lgtm. @masayag what do you think?

TBH, I don't find real value in this PR.
This PR is written as an integration test (uses the standard API of backstage to submit a specific request) via the Kubeapi prebuilt task.
However, this code is never executed. It just "sits" there waiting (or already) to become obsolete since it is never being tested. The same goes for other examples of API usage, without checking the API behaves as expected.

So what's the point of having those tests? It could fit into a README or example doc instead.
Or do an extra effort and move this one to the integration test or add a unit-test with a mock of k8s server to fakely process the tekton CRs.

**Comment by ydayagi** [comment]
it was agreed to close this PR and add something to the docs

**Pull request closed without merging**